htlcswitch/switch.go

```diff
@@ -2994,6 +2994,15 @@ func (s *Switch) handlePacketSettle(packet *htlcPacket) error {
 	// If the source of this packet has not been set, use the circuit map
 	// to lookup the origin.
 	circuit, err := s.closeCircuit(packet)
+
+	// If the circuit is in the process of closing, we will return a nil as
+	// there's another packet handling undergoing.
+	if errors.Is(err, ErrCircuitClosing) {
```

Comment:
Hmm I am not sure if we should return nil here, because we shouldn't call the settle logic twice if we are already closing the circuit and if we do so shouldn't we return an error signaling that the circuit is closing rather than allowing it to be called multiple times and neglecting the error ?

Reply:
I also think we should return an error here and let it be handled - unfortunately we cannot easily do it atm.  I view this as a followup of 8c0c53eac3 - in addition of catching the closed case, we should also catch the closing case. Tho it does feel like the logic should be put inside `closeCircuit`?

Reply:
ahh ok given this fact I think it is ok to place the logic here. But I would recommend to add a comment here why we can end up calling this logic here twice the current comment I would not be able to explain what this really means in detail or how this might happen:

```
there's another packet handling undergoing.
```

i this really only the case if resolution and the normal `htlcPlex` loggic trigger shortly after another ?